iot/spec/iot_spec.rb

```diff
@@ -146,8 +149,8 @@ def create_pubsub_topic topic_id
     )
 
     device_id = "unauth_device"
-    expect {
-      $create_unauth_device.call(
+    expect_with_retry {
```

Comment:
expect_with_retry needs a sample_name argument.

Also, generally I'd prefer the `do ... end` syntax over curly braces for multi-line blocks